azurerm/internal/services/mssql/mssql_database_resource.go

```diff
@@ -388,6 +388,9 @@ func resourceMsSqlDatabaseCreateUpdate(d *schema.ResourceData, meta interface{})
 	}
 
 	if v, ok := d.GetOk("max_size_gb"); ok {
+		if createMode == string(sql.CreateModeOnlineSecondary) || createMode == string(sql.Secondary) {
```

Comment:
Shall we error out in this case, since a warn log here will only show up if the user is running Terraform with TF_LOG specified, which seems not so helpful?
Also, shall we move this checking to the `CustomizeDiff` function instead?

Reply:
Agreed we should probably error out here - but rather than moving this to the CustomizeDiff, we should probably copy it since it's likely the CustomizeDiff will have an interpolated value (from the initial server) - and so may not be able to catch this at that point

Reply:
These two properties here (`max_size_gb` and `create_mode`) seems not likely interpolated from external source. iirc the static configured values can be resolved in diff stage?

Reply:
I've added it to the `CustomizeDiff` and changed the warning to an error

Reply:
I imagine most people would set the size based on the source server - thus it would be interpolated and which makes the customize diss less then idea. the error on create should be enough

Reply:
`CustomizeDiff` is removed